src/main/java/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,183 @@
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to create order "
+                    + order.getId(), e);
+        }
+        setNewProducts(order);
+        return order;
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        Order order = null;
+        String query = "SELECT * FROM orders WHERE order_id = ? AND is_deleted = FALSE";
+        try (Connection conn = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = conn.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                order = getOrder(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to get order by id = "
+                    + id, e);
+        }
+        return Optional.ofNullable(order);
+    }
+
+    @Override
+    public List<Order> getAll() {
+        List<Order> orders = new ArrayList<>();
+        String query = "SELECT * FROM orders WHERE is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Order order = getOrder(resultSet);
+                orders.add(order);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to get all carts!", e);
+        }
+        return orders;
+    }
+
+    @Override
+    public Order update(Order order) {
+        String query = "UPDATE orders SET user_id = ? "
+                + "WHERE is_deleted = FALSE AND order_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, order.getUserId());
+            statement.setLong(2, order.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to update order "
+                    + order.getId(), e);
+        }
+        deleteOldProducts(order);
+        setNewProducts(order);
+        return order;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE orders SET is_deleted = TRUE WHERE order_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to delete order with id = "
+                    + id, e);
+        }
+    }
+
+    private Order getOrder(ResultSet resultSet) throws SQLException {
```

Comment:
maybe better will be getOrderFromResultSet. I think that will be more informative